### PR TITLE
button-group: Fix `btn-group-vertical` by using same rules as `btn-group`

### LINKED ISSUE #40487: Wrong border radius first element in `btn-group-vertical` is a dropdown
### Describe the issue

With horizontal button groups, the first button can be a dropdown and the border radius looks like expected:

> image

With vertial button groups, the border radius breaks when the first item of the list is a dropdown:

> image

### Reduced test cases

https://stackblitz.com/edit/1vpwjr?file=index.html

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

https://getbootstrap.com/docs/5.2/components/button-group/

### PR DESCRIPTION
**Pull request by tordans**
### Description

The CSS conditions for vertical button groups was different from the regular button group.

When using the same conditions for the regular groups for the vertical group, the bug is fixed.

**Current CSS:**

See the `.btn-group>:not(.btn-check)+.btn` on the right

> Bildschirmfoto 2024-05-25 um 07 06 39

**Modified CSS:**

> Bildschirmfoto 2024-05-25 um 07 08 03

I am assuming the css for the regular button group was improved at some point but not applied to the vertical button group at this point.

~The second change is to add `> .btn.dropdown-toggle-split:first-child` to the `Reset rounded corners` section which is also copied down from the regular button group. I am not sure what that does but I think the same logic applies to assume that both button groups should behave the same.~ – Update: I removed this again because the docs state that the split buttons are not supported for vertical button groups.

I have modified the docs page of the second example on https://getbootstrap.com/docs/5.2/components/button-group/#vertical-variation to have a button group in the first position.

### Type of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] ~New feature (non-breaking change which adds functionality)~
- [ ] ~Refactoring (non-breaking change)~
- [ ] ~Breaking change (fix or feature that would change existing functionality)~

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [ ] ~I have updated the documentation accordingly~
- [ ] ~I have added tests to cover my changes~
- [x] All new and existing tests passed

#### Live previews

- https://deploy-preview-40488--twbs-bootstrap.netlify.app/docs/5.3/components/button-group

### Related issues

- Closes https://github.com/twbs/bootstrap/issues/40487

### CODE REVIEW

**Review by julien-deramond — CHANGES_REQUESTED** [review]
Thanks for catching the issue in the first place and then for submitting a PR @tordans 

I haven't checked out the code yet, but there's an issue with the rendering in the deployed version at https://deploy-preview-40488--twbs-bootstrap.netlify.app/docs/5.3/components/button-group/#vertical-variation; the top-right corner from the 2nd item in the vertical variation is rounded, and the left-bottom corner of the last item should be rounded.

Screenshot 2024-05-25 at 07 53 00
Screenshot 2024-05-25 at 07 52 55

File: scss/_button-group.scss [hunk 1]
```diff
@@ -131,12 +131,18 @@
...
-  > .btn ~ .btn,
+  // The top radius should be 0 if the button is:
+  // - the "third or more" child
+  // - the second child and the previous element isn't `.btn-check` (making it the first child visually)
+  // - part of a btn-group which isn't the first child
+  > .btn:nth-child(n + 3),
+  > :not(.btn-check) + .btn,
   > .btn-group:not(:first-child) > .btn {
-    @include border-top-radius(0);
+    @include border-start-radius(0);
```
Comment (tordans) [inline scss/_button-group.scss:146]:
```suggestion
    @include border-top-radius(0);
```

**Review by tordans — COMMENTED** [review]
This change was not supposed to be in here. Will fix soon.

**Comment by PRERAKAGARWAL28** [comment]
Thanks for catching the issue in the first place and then for submitting a PR @tordans
I have resolved the issue. 
https://stackblitz.com/edit/jzsfzg?file=index.html

**Comment by tordans** [comment]
> I haven't checked out the code yet, but there's an issue with the rendering in the deployed version at https://deploy-preview-40488--twbs-bootstrap.netlify.app/docs/5.3/components/button-group/#vertical-variation; the top-right corner from the 2nd item in the vertical variation is rounded, and the left-bottom corner of the last item should be rounded.

This is fixed now. The preview looks right to me, now.

**Review by julien-deramond — APPROVED** [review]
Thanks again for the PR, @tordans. It's really appreciated.

I've tested the rendering with several combinations of buttons and dropdowns, and it seems to work perfectly now. Even with some outlined versions, the rendering is acceptable. The following mention in the documentation is still valid, even if it shouldn't be too hard to adapt in the future:
> Split button dropdowns are not supported here.

The comment follows what exists already for the simple `.btn-group`, and the new code is scoped for `.btn-group-vertical` only. This makes it very isolated, without extra impacts on the rest of the code.

Regarding the documentation, having the dropdown in the first position will make it easier for us to maintain in the future and to spot potential regressions. We do lose a little bit of the logic behind the rendering of the set of dropdowns' variants (with two buttons in the middle), but I think it's OK. Otherwise, we could even drop the buttons. IMO, let's leave it as is. If other maintainers have stronger opinions, feel free to directly modify the code.

IMO, it's safe to merge. @twbs/css-review, I'll leave it open for now until you can double-check if you want.

**Review by louismaximepiton — APPROVED** [review]
Feels right to me, couldn't find any regression while playing with it. Thanks for your contribution!

**Pull request merged**